### PR TITLE
Update java pubsub bot to latest cloud client lib, simplify code

### PR DESCRIPTION
**Pull request by sqrrrl**

### CODE REVIEW

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 1]
```diff
@@ -13,26 +13,22 @@
...
-import com.fasterxml.jackson.databind.node.ObjectNode;
-import com.google.api.client.googleapis.auth.oauth2.GoogleCredential;
 import com.google.api.client.googleapis.javanet.GoogleNetHttpTransport;
-import com.google.api.client.http.ByteArrayContent;
-import com.google.api.client.http.GenericUrl;
-import com.google.api.client.http.HttpContent;
-import com.google.api.client.http.HttpRequest;
-import com.google.api.client.http.HttpRequestFactory;
-import com.google.api.client.http.HttpTransport;
+import com.google.api.client.http.*;
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: No star imports please https://google.github.io/styleguide/javaguide.html#s3.3.1-wildcard-imports

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 2]
```diff
@@ -41,129 +37,91 @@
  */
 public class Main {
 
-  public static final String CREDENTIALS_PATH_ENV_PROPERTY = "GOOGLE_APPLICATION_CREDENTIALS";
-
-  // Google Cloud Project ID
-  public static final String PROJECT_ID = "YOUR-PROJECT-ID";
+  public static void main(String[] args) {
+    System.out.println(System.getenv().toString());
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: does need to be printed out? add an info log if valuable

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 3]
```diff
@@ -41,129 +37,91 @@
  */
 public class Main {
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: Can you rename the file to PubSubBot.java?

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 4]
```diff
@@ -41,129 +37,91 @@
...
+                .setApplicationName("pubsub-bot-java")
+                .build();
+
+        chatService.spaces().messages().create(spaceName, reply).execute();
+        consumer.ack();
+      } catch (Exception e) {
+        e.printStackTrace();
+        consumer.nack();
+      }
+    }
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: nit on new line at EOF

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 5]
```diff
@@ -41,129 +37,91 @@
...
-    String eventType = eventJson.get("type").asText();
-    switch (eventType) {
-      case "ADDED_TO_SPACE":
-        responseNode.put("text", ADDED_RESPONSE);
-        // A bot can also be added to a room by @mentioning it in a message. In that case, we fall
-        // through to the MESSAGE case and let the bot respond. If the bot was added using the
-        // invite flow, we just post a thank you message in the space.
-        if(!eventJson.has("message")) {
-          break;
+    // Called when a message is received by the subscriber.
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: please add a javadoc

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 6]
```diff
@@ -41,129 +37,91 @@
...
+    @Override
+    public void receiveMessage(PubsubMessage pubsubMessage, AckReplyConsumer consumer) {
+      System.out.println("Id : " + pubsubMessage.getMessageId());
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: remove

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 7]
```diff
@@ -41,129 +37,91 @@
...
+      // handle incoming message, then ack/nack the received message
+      try {
+        String content = pubsubMessage.getData().toStringUtf8();
+        ObjectMapper mapper = new ObjectMapper();
+        JsonNode chatEvent = mapper.readTree(content);
+
+        Message reply = null;
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: let's refactor to later check a boolean or initialize to an empty message and check if content present later

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 8]
```diff
@@ -41,129 +37,91 @@
...
+        }
 
-    HttpContent content =
-        new ByteArrayContent("application/json", responseNode.toString().getBytes("UTF-8"));
-    HttpRequest request = requestFactory.buildPostRequest(url, content);
-    com.google.api.client.http.HttpResponse response = request.execute();
+        // Send the reply message via chat API.
+        String spaceName = chatEvent.at("/space/name").asText();
+        GoogleCredentials credentials = GoogleCredentials.getApplicationDefault().createScoped(
+                "https://www.googleapis.com/auth/chat.bot"
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: add earlier as a constant

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java [hunk 9]
```diff
@@ -41,129 +37,91 @@
...
+        );
+        HttpRequestInitializer requestInitializer = new HttpCredentialsAdapter(credentials);
+        HangoutsChat chatService = new HangoutsChat.Builder(
+                GoogleNetHttpTransport.newTrustedTransport(),
+                JacksonFactory.getDefaultInstance(),
+                requestInitializer)
+                .setApplicationName("pubsub-bot-java")
```
Comment (asrivas) [inline java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: add earlier as a constant

File: java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java (hunk 2 above)
Comment (sqrrrl) [reply java/pub-sub-bot/src/main/java/com/google/chat/bot/pubsub/Main.java]: Oops, was just left over from some quick debugging :)

**Pull request merged**